### DISCUSSION TITLE
Support for cross-run RDMs

### DISCUSSION THREAD

**Discussion by El-Suri** [rsagroup/rsatoolbox#282] [General]
Hello,

I am running an analysis using only stimulus comparisons across different fMRI runs (a cross-run RDM). At the moment I am calculating an RDM on the entire dataset and then am manually going in to select only the across-run comparisons. As I have 3 runs, this results in 3 RDMs which I then collapse into a single RDM by taking the mean of these.

This works well enough, however when I then put this new cross-run RDM into an rsatoolbox RDM object, it automatically makes the diagonal 0's and makes the matrix symmetrical either side. This is of course normal for a within-run or full RDM, however in this particular case this is not accurate as it is not the exact same-stimulus being compared on the diagonal. In addition, the comparisons are not symmetrical. For example, stimus1 on the x axis is not the same as stimulus1 on the y axis because they are in fact now simply means representing different comparisons across runs. This would also be the case if I was not taking a mean, for example by selecting only run1 vs run2 comparisons. 

Perhaps this isn't a major problem, however I am concerned that by making the RDMs symmetrical and introducing the 0's on the diagonal, this is reducing the amount of datapoints I have. I assume this is done because all of the analyses on the RDMs further downstream are performed on the upper triangle only. Seeing as this is not necessary for cross-run RDMs however, perhaps it could be an option to perform analyses on entire matrices and not have the rdm object automatically introduce symmetry and 0's along the diagonal? 

Thoughts are most welcome, I have included part of a recent lab presentation which hopefully explains what I mean more clearly. 

![Screenshot 2022-11-09 at 14 22 00](https://user-images.githubusercontent.com/34473937/200854909-3217f6b5-c339-455d-966b-cd08e7cc7840.png)

![Screenshot 2022-11-09 at 14 22 09](https://user-images.githubusercontent.com/34473937/200854920-ed4f6320-b784-4160-93a1-75dd11091cf1.png)

![Screenshot 2022-11-09 at 14 20 49](https://user-images.githubusercontent.com/34473937/200854592-1b29e60d-f104-42b4-9a12-67bb76553e2d.png)

![Screenshot 2022-11-09 at 14 21 01](https://user-images.githubusercontent.com/34473937/200854639-869377eb-95f9-4167-8554-8395a2b748b4.png)

**Comment by JasperVanDenBosch** [comment]
Hi @El-Suri, thanks for raising this. I'm gonna turn this into a Github Discussion, as this could be a longer conversation.

**Comment by JasperVanDenBosch** [comment]
> Within-run stimulus comparisons are not okay in our case..

Not a full answer, but maybe [this paper](https://www.biorxiv.org/content/10.1101/032391v2.abstract) could be relevant for this.
